servers/src/main/java/tachyon/worker/block/allocator/RoundRobinAllocator.java

```diff
@@ -0,0 +1,120 @@
+    
+    for (StorageTierView tierView : mManagerView.getTierViews()) {
+      mTierViewDirViews.put(tierView, -1);
+    }
+  }
+
+  @Override
+  public TempBlockMeta allocateBlockWithView(long userId, long blockId, long blockSize,
+      BlockStoreLocation location, BlockMetadataManagerView view) throws IOException {
+    mManagerView = view;
+    return allocateBlock(userId, blockId, blockSize, location);
+  }
+
+  /**
+   * Should only be accessed by {@link allocateBlockWithView} inside class.
+   * Allocates a block from the given block store location. The location can be a specific location,
+   * or {@link BlockStoreLocation#anyTier()} or {@link BlockStoreLocation#anyDirInTier(int)}.
+   *
+   * @param userId the ID of user to apply for the block allocation
+   * @param blockId the ID of the block
+   * @param blockSize the size of block in bytes
+   * @param location the location in block store
+   * @return a temp block meta if success, null otherwise
+   * @throws IOException if block location is invalid
+   */
+  private TempBlockMeta allocateBlock(long userId, long blockId, long blockSize,
+      BlockStoreLocation location) throws IOException {
+    if (location.equals(BlockStoreLocation.anyTier())) {
+      int tierAlias = 0; //always starting from the first tier
+      for (int i = 0; i < mManagerView.getTierViews().size(); i ++) {
+        StorageTierView tierView = mManagerView.getTierViews().get(tierAlias);
+        int dirViewIndex = getNextDirViewInTierView(tierView, blockSize);
+        if (dirViewIndex >= 0) {
+          mTierViewDirViews.put(tierView, dirViewIndex); // update
+          return tierView.getDirView(dirViewIndex).createTempBlockMeta(userId, blockId, blockSize);
+        } else { // we didn't find one in this tier, go to next tier
+          tierAlias ++;
+        }
+      }
+    } else if (location.equals(BlockStoreLocation.anyDirInTier(location.tierAlias()))) {
+      StorageTierView tierView = mManagerView.getTierView(location.tierAlias());
+      int dirViewIndex = getNextDirViewInTierView(tierView, blockSize);
+      if (dirViewIndex >= 0) {
+        mTierViewDirViews.put(tierView, dirViewIndex); // update
+        return tierView.getDirView(dirViewIndex).createTempBlockMeta(userId, blockId, blockSize);
+      }
+    } else {
+      StorageTierView tierView = mManagerView.getTierView(location.tierAlias());
+      StorageDirView dirView = tierView.getDirView(location.dir());
+      if (dirView.getAvailableBytes() >= blockSize) {
+        return dirView.createTempBlockMeta(userId, blockId, blockSize);
+      }
+    }
+
+    return null;
+  }
+
+  /**
+   * Find an available dir in a given tier for a block with blockSize
+   * @param tier: the tier to find a dir
+   * @param blockSize: the requested block size
+   * @return: the index of the dir if nonnegative; -1 if fail to find a dir
+   */
+  private int getNextDirViewInTierView(StorageTierView tierView, long blockSize) 
```

Comment:
You should name it to emphasize that it finds an available view, not just a next view. Also you can leave out `view` in the method name. We are always assuming using views anyway.